symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/util/MessageParser.java

```diff
@@ -0,0 +1,138 @@
+package com.symphony.bdk.core.service.message.util;
+
+import com.symphony.bdk.core.service.message.exception.MessageParserException;
+import com.symphony.bdk.gen.api.model.V4Message;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class for extracting entities inside an incoming {@link com.symphony.bdk.gen.api.model.V4Message} such as:
+ * Mentions, Hashtags, Cashtags, Emojis.
+ */
+@Slf4j
+@API(status = API.Status.EXPERIMENTAL)
+public class MessageParser {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private MessageParser() {
+  }
```

Comment:
Lombok style `@NoArgsConstructor(access = AccessLevel.PRIVATE)`

Reply:
added in each util class